deepchecks/nlp/text_data.py

```diff
@@ -0,0 +1,583 @@
+    index: t.Optional[t.Sequence[int]], default: None
+        The index of the samples. If None, the index is set to np.arange(len(raw_text)).
+    """
+
+    _dataset: TTextDataset
+    _task_type: t.Optional[TaskType]
+    _has_label: bool
+    _label_name: str
+    _is_multilabel: bool = False
+    _classes: t.Optional[t.List[t.Union[str, int]]] = None
+    name: t.Optional[str] = None
+
+    def __init__(
+            self,
+            raw_text: t.Sequence[str],
+            label: t.Optional[TTextLabel] = None,
+            task_type: str = None,
+            label_name: t.Optional[str] = None,
+            dataset_name: t.Optional[str] = None,
+            index: t.Optional[t.Sequence[int]] = None,
+    ):
+
+        # Require explicitly setting task type if label is provided
+        if task_type is None:
+            if self.label is not None:
+                raise DeepchecksValueError('task_type must be set when label is provided')
+            self._task_type = None
+        elif task_type == 'text_classification':
+            self._task_type = TaskType.TEXT_CLASSIFICATION
+        elif task_type == 'token_classification':
+            self._task_type = TaskType.TOKEN_CLASSIFICATION
+        else:
+            raise DeepchecksNotSupportedError(f'task_type {task_type} is not supported, must be one of '
+                                              f'text_classification, token_classification')
+
+        # Set label name to default if not provided
+        if label_name is None:
+            self._label_name = DEFAULT_LABEL_NAME
+        elif isinstance(label_name, (str, float, int)):
+            self._label_name = label_name
+        else:
+            raise DeepchecksNotSupportedError(f'label_name {label_name} is not supported, must be one of '
+                                              f'str, float, int')
+
+        self._validate_text(raw_text)
+        label = self._validate_and_process_label(label, raw_text)
+
+        if index is None:
+            index = np.arange(len(raw_text))
+        elif len(index) != len(raw_text):
+            raise DeepchecksValueError('index must be the same length as raw_text')
+
+        # Create dataset
+        self._dataset = {'text': raw_text, self._label_name: label, 'index': index}
+
+        # Set dataset name
+        if dataset_name is None:
+            self.name = f'text_{self._label_name}'
+        else:
+            self.name = dataset_name
+
+        if dataset_name is not None:
+            if not isinstance(dataset_name, str):
+                raise DeepchecksNotSupportedError(f'dataset_name {dataset_name} is not supported, must be a str')
+        self.name = dataset_name
+
+    @staticmethod
+    def _validate_text(raw_text: t.Sequence[str]):
+        """Validate text format."""
+        if not isinstance(raw_text, collections.abc.Sequence):
+            raise DeepchecksValueError('raw_text must be a sequence')
+        if not all(isinstance(x, str) for x in raw_text):
```

Comment:
maybe just convert to str? edge cases of char / sequeance of chars/ number

Reply:
+1 for bressler